runtime/observability/telemetry-defaults/common/src/aws/smithy/kotlin/runtime/telemetry/GlobalTelemetryProvider.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.telemetry
+
+import aws.smithy.kotlin.runtime.telemetry.context.ContextManager
+import aws.smithy.kotlin.runtime.telemetry.logging.DefaultLoggerProvider
+import aws.smithy.kotlin.runtime.telemetry.logging.LoggerProvider
+import aws.smithy.kotlin.runtime.telemetry.metrics.MeterProvider
+import aws.smithy.kotlin.runtime.telemetry.trace.TracerProvider
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+
+/**
+ * Global [TelemetryProvider]
+ */
+public object GlobalTelemetryProvider {
```

Comment:
**Style**: I don't like calling this the "global" telemetry provider. That implies it will be used everywhere (i.e., globally) when in fact it's merely the default telemetry provider used when callers don't specify their own. I suggest its name should be `DefaultTelemetryProvider`, accessible via `TelemetryProvider.Companion.Default`. I think the actual implementation could be named something like `DefaultTelemetryProviderImpl`, `StandardTelemetryProdivder`, etc.

Reply:
My issue is entirely with the name itself. The idea of setting a default used by future, disconnected client initializers makes sense to me and I can see how we might want to do it in other places (e.g., default credentials provider/chain).

Reply:
I can live with a name change. The reason I named it "Global" though was it is slightly different from all our other usage of "DefaultXyz". The "Global" part of the name also immediately signals the scope of this thing which I find useful.